spec/components/step_by_step_nav_spec.rb

```diff
@@ -165,7 +165,7 @@ def stepnav
 
     assert_select step1 + " #step-panel-step-1-1.gem-c-step-nav__panel"
     assert_select step1and + " #step-panel-step-1-and-2.gem-c-step-nav__panel"
-    assert_select step3 + "#step-3"
+    assert_select step3 + "#step-3-_"
```

Comment:
Not sure I understand this change, can you clarify?

Reply:
This is where the parameterize method differs.  It won't replace underscores, so we end up with the original underscore and a hyphen joining it to the preceding "word".  

It looks a bit weird, but I didn't want to fudge the test, and it's a good demonstration of what it does.  In reality, I don't think we'd ever have a title with underscores in it, so the weird looking URL wouldn't occur.

There's nothing functionally wrong with this btw, it just looks odd

Reply:
It's generated from`Step 3?"`_!`